pymc_marketing/mmm/events.py

```diff
@@ -0,0 +1,207 @@
+        }
+
+        return self._sample_curve(
+            var_name="saturation",
+            parameters=parameters,
+            x=x,
+            coords=coords,
+        )
+
+
+def basis_from_dict(data: dict) -> Basis:
+    """Create a basis transformation from a dictionary."""
+    data = data.copy()
+    lookup_name = data.pop("lookup_name")
+    cls = BASIS_TRANSFORMATIONS[lookup_name]
+
+    if "priors" in data:
+        data["priors"] = {k: deserialize(v) for k, v in data["priors"].items()}
+
+    return cls(**data)
+
+
+def _is_basis(data):
+    return "lookup_name" in data and data["lookup_name"] in BASIS_TRANSFORMATIONS
+
+
+register_deserialization(
+    is_type=_is_basis,
+    deserialize=basis_from_dict,
+)
+
+
+@dataclass
+class EventEffect:
+    basis: Basis
+    effect_size: Prior
+    dims: tuple[str, ...]
+
+    def apply(self, X, name: str = "event"):
+        """Apply the event effect to the data."""
+        dim_handler = create_dim_handler(("x", *self.dims))
+        return self.basis.apply(X, dims=self.dims) * dim_handler(
+            self.effect_size.create_variable(f"{name}_effect_size"),
+            self.effect_size.dims,
+        )
+
+    def to_dict(self):
+        return {
+            "class": "EventEffect",
+            "data": {
+                "basis": self.basis.to_dict(),
+                "effect_size": self.effect_size.to_dict(),
+                "dims": self.dims,
+            },
+        }
+
+    @classmethod
+    def from_dict(cls, data):
+        return cls(
+            basis=deserialize(data["basis"]),
+            effect_size=deserialize(data["effect_size"]),
+            dims=data["dims"],
+        )
+
+
+def _is_event_effect(data):
+    return data["class"] == "EventEffect"
+
+
+register_deserialization(
+    is_type=_is_event_effect,
+    deserialize=lambda data: EventEffect.from_dict(data["data"]),
+)
+
+
+if __name__ == "__main__":
+    import matplotlib.pyplot as plt
+
+    from pymc_marketing.plot import plot_curve
+
+    class GaussianBasis(Basis):
+        lookup_name = "gaussian"
+
+        def function(self, x, sigma):
+            return pm.math.exp(-0.5 * (x / sigma) ** 2)
+
+        default_priors = {
+            "sigma": Prior("Gamma", mu=7, sigma=1),
+        }
+
+    gaussian = GaussianBasis(
+        priors={
+            "sigma": Prior("Gamma", mu=[4, 7, 10], sigma=1, dims="event"),
+        },
+    )
+    coords = {"event": ["NYE", "Grand Opening Game Show", "Super Bowl"]}
+    prior = gaussian.sample_prior(coords=coords)
+    curve = gaussian.sample_curve(prior, days=21)
+
+    fig, axes = gaussian.plot_curve(curve, same_axes=True)
+    fig.suptitle("Gaussian Basis")
+    plt.savefig("gaussian-basis")
+    plt.close()
+
+    df_events = pd.DataFrame(
+        {
+            "event": ["first", "second"],
+            "start_date": pd.to_datetime(["2023-01-01", "2023-01-20"]),
+            "end_date": pd.to_datetime(["2023-01-02", "2023-01-25"]),
+        }
+    )
```

Comment:
This would be the structure that people would add to their MMM

Would this be passed at init or at some other point. It would have to be before `build_model` somehow in order to intercept the mu variable